SConstruct

```diff
@@ -296,6 +334,7 @@ elif env["platform"] == "windows":
         # Don't Clone the environment. Because otherwise, SCons will pick up msvc stuff.
         env = Environment(ENV=os.environ, tools=["mingw"])
         opts.Update(env)
+        env["arch"] = env_arch
         # env = env.Clone(tools=['mingw'])
```

Comment:
this could share a common conditional for windows (from above), just to make it a little bit clear

Reply:
I'm not sure what you mean, but we need to set `env["arch"] = env_arch` any time we call `opts.Update(env)` since the latter will update the environment with the command line arguments.

Reply:
My point was a real minor detail about the implementation. Like nesting the if statement  nothing important..
```
if host_platform == "windows":
    if env["use_mingw"]:
        # some stuffs to do
    else:
        # etc
```

However I will pay attention to the other comment and your answer because I thought someone was missing I hope I am wrong. First I will have to understand before any conclusion obviously.
Thanks for the clarification, I'm trying to learn how SCons is used in godot. Ok it's late, thank you! Later